### PR TITLE
Add jenkins

### PR DESCRIPTION
**Pull request by FridaTveit**
Jenkinsfile will:
* Dotnet build/test
* Typescript build/test/lint
* Notify slack

To run the jenkinsfile on windows sh would need to be replaced with bat. For this part of the workshop I think it would be useful to have some instructions in the model answer, as the output is the jenkins job just as much as the jenkinsfile. I'd suggest adding that here as a markdown file with some instructions and screenshots (I'd do that as a separate MR). Let me know if you have any concerns about that or have a different suggestion 🙂

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+    }
+
+    post {
+        failure {
+            slack(":red_circle: Build failed", "danger")
+        }
+        success {
+            slack(":excellent: Build succeeded", "good")
```
Comment (thehereward) [inline Jenkinsfile:49]: Unfortunately `:excellent:` isn't included in Slack by default.
```suggestion
            slack(":heavy_check_mark: Build succeeded", "good")
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            }
+        }
+        stage('Typescript Tests') {
+            steps {
+                dir ("DotnetTemplate.Web/") {
+                    sh "npm test"
+                }
+            }
+        }
+        stage('Typescript Lint') {
```
Comment (thehereward) [inline Jenkinsfile:35]: Isn't linting a relatively quick step?
Wouldn't it be better to put it earlier?

Comment (FridaTveit) [reply Jenkinsfile:35]: Yes it is and sure I could move it earlier. What do you think would be the advantage of that? 🙂

Comment (thehereward) [reply Jenkinsfile:35]: To make the build fail faster; don't bother doing anything time consuming if there's a semi-colon in the wrong place (for example).

Comment (FridaTveit) [reply Jenkinsfile:35]: Fair enough, happy to move it around if you prefer

**Review by thehereward — APPROVED** [review]

**Pull request merged**